src/core/device.c

```diff
@@ -766,6 +766,28 @@ static int device_dispatch_io(sd_event_source *source, int fd, uint32_t revents,
                 return 0;
         }
 
+        if (streq(action, "change"))  {
+                _cleanup_free_ char *e = NULL;
+                Unit *u;
+
+                r = unit_name_from_path(sysfs, ".device", &e);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to generate unit name from device path: %m");
+
+                u = manager_get_unit(m, e);
+                if (u && UNIT_VTABLE(u)->active_state(u) == UNIT_ACTIVE) {
+                        r = manager_propagate_reload(m, u, JOB_REPLACE, NULL);
+                        if (r < 0) {
+                                log_error_errno(r, "Failed to add job to manager: %m");
```

Comment:
`"Failed to propagate reload: %m"` maybe?

Reply:
I don't know what kind of user-side error could reach that point... this is probably an out-of-memory or some harder to diagnose condition. 

But yes, your error message is slightly better, i'll wait for a couple more review and push a new version